build/helper/documentation_helper.py

```diff
@@ -402,7 +402,7 @@ def _format_type_for_docstring(param, numpy, config):
 
 
 rep_cap_method_desc_docstring = rep_cap_method_desc + '''
-    session['0,1'].{1}({2})
+    session.channel[[0, 1]].{1}({2})
```

Comment:
Question for the Python crowd... is indexing using an array "Pythonic"? Or is it strange?

Reply:
Channels 0, 1, 4, 12 definitely needs to be supported. Something like `my_channels = session.channels[0:1] + session.channels[4] + session.channels[12]` to do it though seems like a step backwards.

Reply:
If you wanted a list that contains elements at indices 0, 1, 4, and 12, how would you construct that list?

Reply:
Indexing using arrays is not strange at all. See Numpy's advanced indexing feature: https://docs.scipy.org/doc/numpy-1.13.0/reference/arrays.indexing.html#advanced-indexing

Reply:
We don't want 4 object in a list out of this. We need one object where we can internally put together a string that looks like '0,1,4,12'.

Reply:
@texasaggie97 So do that: index with an array and return whatever object you need.

Reply:
After seeing numpy's "advanced-indexing" I think allowing an array makes sense. I think as long as the user can use integers and slices then users will find themselves successful. Allowing arrays seems like a "nice-to-have" but if it is easy to support, and the user can easily understand what happens when an array is used, then I say go for it.

> We need one object where we can internally put together a string that looks like '0,1,4,12'.
The string is an implementation detail. What you want is for the user to be able to say "on channels x do y". Since we're exposing the "seuqence" version of  `__getitem__` on the `channels` attribute to access specific channels, I see it the same as accessing indices of a sequence type to operate on.

Reply:
Yes, it is an implementation detail, but it is an important detail. We need the call to condense down to a single call with '0,1,4,12' not a call with '0,1', then a call with '4', then a call with '12'.

It is very possible that I don't see something that would allow `my_channels = session.channels[0:1] + session.channels[4] + session.channels[12]` to do this. But also, this seems  very verbose.

`(session.channels[0:1] + session.channels[4] + session.channels[12]).fetch(1000)` seems unwieldy.

Reply:
This can probably be achieved, if we decide that's the way we want to go, by implementing __add__().

Reply:
> The string is an implementation detail.

Under the hood, this will touch hardware. If a user sets the voltage on several channels "all at once" but the Python code somewhat serializes this, DUT can be damaged. Mark is right that we need to make this work.

Reply:
I agree serializing the calls is not an option. I just want to make sure the implementation details don't bleed out into the interface. After seeing that numpy uses arrays to mean a list of indices, this is starting to look less weird.

I also don't think `add()` should be added to the type that `__getitem__()` returns just yet (mainly because the list-of-indices should cover *most* of the use cases). I just wanted to bring it up to show the similarities between two sequence types (yours and lists), to help us think about how we should design the interface,

Just make sure the normal use case isn't broken (integers and slices), and the array-of-indices has good documentation and I think you're good to go. 👍